### PR TITLE
Query History: Add feature tracking for query history

### LINKED ISSUE #49668: Query History: Feature tracking
Track:
* Opening query history panel
* Running a query from the history (with attributes indicating if the data source was switched)
* Starring
* Commenting
* Deleting

All events should contain an attribute to indicate if new query history is enabled.

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

To make better decisions about further improvements we need to start tracking usage of query history.

**Which issue(s) this PR fixes**:

Fixes #49668

**Special notes for your reviewer**:

How to test it?

Add the following line in `public/app/core/services/echo/utils.ts:20` to see events tracked in the console:

```diff
 /** @internal */ 
 export const echoLogger = createLogger('EchoSrv');
 export const echoLog = echoLogger.logger;
+echoLogger.enable();

 attachDebugger('echo', undefined, echoLogger);

```

### CODE REVIEW

File: public/app/features/explore/RichHistory/RichHistoryCard.tsx [hunk 1]
```diff
@@ -176,12 +176,17 @@ export function RichHistoryCard(props: Props) {
...
     const queriesToRun = query.queries;
-    if (query.datasourceName !== datasourceInstance?.name) {
+    const differentDataSource = query.datasourceName !== datasourceInstance?.name;
+    if (differentDataSource) {
       await changeDatasource(exploreId, query.datasourceName, { importQueries: true });
       setQueries(exploreId, queriesToRun);
     } else {
       setQueries(exploreId, queriesToRun);
     }
+    reportInteraction('explore_query_history_run', {
```
Comment (ivanahuckova) [inline public/app/features/explore/RichHistory/RichHistoryCard.tsx:186]:
```suggestion
    reportInteraction('grafana_explore_query_history_run', {
```

File: public/app/features/explore/RichHistory/RichHistoryCard.tsx [hunk 2]
```diff
@@ -196,6 +201,14 @@ export function RichHistoryCard(props: Props) {
   };
 
   const onDeleteQuery = () => {
+    const performDelete = (queryId: string) => {
+      deleteHistoryItem(queryId);
+      dispatch(notifyApp(createSuccessNotification('Query deleted')));
+      reportInteraction('explore_query_history_deleted', {
```
Comment (ivanahuckova) [inline public/app/features/explore/RichHistory/RichHistoryCard.tsx:207]:
```suggestion
      reportInteraction('grafana_explore_query_history_deleted', {
```

File: public/app/features/explore/RichHistory/RichHistoryCard.tsx [hunk 3]
```diff
@@ -204,27 +217,30 @@ export function RichHistoryCard(props: Props) {
...
     } else {
-      deleteHistoryItem(query.id);
-      dispatch(notifyApp(createSuccessNotification('Query deleted')));
+      performDelete(query.id);
     }
   };
 
   const onStarrQuery = () => {
     starHistoryItem(query.id, !query.starred);
+    reportInteraction('explore_query_history_starred', {
```
Comment (ivanahuckova) [inline public/app/features/explore/RichHistory/RichHistoryCard.tsx:230]:
```suggestion
    reportInteraction('grafana_explore_query_history_starred', {
```

File: public/app/features/explore/RichHistory/RichHistoryContainer.tsx [hunk 4]
```diff
@@ -86,6 +87,9 @@ export function RichHistoryContainer(props: Props) {
 
   useEffect(() => {
     initRichHistory();
+    reportInteraction('explore_query_history_opened', {
```
Comment (ivanahuckova) [inline public/app/features/explore/RichHistory/RichHistoryContainer.tsx:90]:
```suggestion
    reportInteraction('grafana_explore_query_history_opened', {
```

File: public/app/features/explore/spec/queryHistory.test.tsx [hunk 5]
```diff
@@ -103,6 +108,9 @@ describe('Explore: Query History', () => {
     // previously added query is in query history
     await openQueryHistory();
     await assertQueryHistoryExists(RAW_QUERY);
+
+    expect(reportInteractionMock).toBeCalledTimes(2);
+    expect(reportInteractionMock).toBeCalledWith('explore_query_history_opened', { queryHistoryEnabled: false });
```
Comment (ivanahuckova) [inline public/app/features/explore/spec/queryHistory.test.tsx:113]:
```suggestion
    expect(reportInteractionMock).toBeCalledWith('grafana_explore_query_history_opened', { queryHistoryEnabled: false });
```

File: public/app/features/explore/spec/queryHistory.test.tsx [hunk 6]
```diff
@@ -153,10 +161,15 @@ describe('Explore: Query History', () => {
     starQueryHistory(1, ExploreId.left);
     await assertQueryHistoryIsStarred([false, true], ExploreId.left);
     await assertQueryHistoryIsStarred([false, true], ExploreId.right);
+    expect(reportInteractionMock).toBeCalledWith('explore_query_history_starred', {
```
Comment (ivanahuckova) [inline public/app/features/explore/spec/queryHistory.test.tsx:164]:
```suggestion
    expect(reportInteractionMock).toBeCalledWith('grafana_explore_query_history_starred', {
```

File: public/app/features/explore/spec/queryHistory.test.tsx [hunk 7]
```diff
@@ -153,10 +161,15 @@ describe('Explore: Query History', () => {
...
+      queryHistoryEnabled: false,
+      newValue: true,
+    });
 
     deleteQueryHistory(0, ExploreId.left);
     await assertQueryHistory(['{"expr":"query #1"}'], ExploreId.left);
     await assertQueryHistory(['{"expr":"query #1"}'], ExploreId.right);
+    expect(reportInteractionMock).toBeCalledWith('explore_query_history_deleted', { queryHistoryEnabled: false });
```
Comment (ivanahuckova) [inline public/app/features/explore/spec/queryHistory.test.tsx:172]:
```suggestion
    expect(reportInteractionMock).toBeCalledWith('grafana_explore_query_history_deleted', { queryHistoryEnabled: false });
```

File: public/app/features/explore/spec/queryHistory.test.tsx [hunk 8]
```diff
@@ -192,6 +205,7 @@ describe('Explore: Query History', () => {
 
       await openQueryHistory();
       expect(postMock).not.toBeCalledWith('/api/query-history/migrate', { queries: [] });
+      expect(reportInteractionMock).toBeCalledWith('explore_query_history_opened', { queryHistoryEnabled: false });
```
Comment (ivanahuckova) [inline public/app/features/explore/spec/queryHistory.test.tsx:208]:
```suggestion
      expect(reportInteractionMock).toBeCalledWith('grafana_explore_query_history_opened', { queryHistoryEnabled: false });
```

File: public/app/features/explore/spec/queryHistory.test.tsx [hunk 9]
```diff
@@ -219,6 +233,7 @@ describe('Explore: Query History', () => {
           url: expect.stringMatching('/api/query-history/migrate'),
         })
       );
+      expect(reportInteractionMock).toBeCalledWith('explore_query_history_opened', { queryHistoryEnabled: true });
```
Comment (ivanahuckova) [inline public/app/features/explore/spec/queryHistory.test.tsx:236]:
```suggestion
      expect(reportInteractionMock).toBeCalledWith('grafana_explore_query_history_opened', { queryHistoryEnabled: true });
```

**Review by ivanahuckova — COMMENTED** [review]
👼 the naming should start with the product -> `grafana`. Otherwise LGTM.

File: public/app/features/explore/spec/queryHistory.test.tsx (hunk 7 above)
Comment (ivanahuckova) [reply public/app/features/explore/spec/queryHistory.test.tsx:172]: Also for interactions with specific queries, I am wondering if we shouldn't have `grafana_explore_query_history_query_deleted`. But not sure. Just to distinguish between interaction with query and whole query history.

**Review by gelicia — APPROVED** [review]

File: public/app/features/explore/spec/queryHistory.test.tsx (hunk 7 above)
Comment (ifrost) [reply public/app/features/explore/spec/queryHistory.test.tsx]: I'll merge it like this. I've added `grafana_` prefix 👍

**Pull request merged**